std/json.d

```diff
@@ -1725,8 +1727,11 @@ pure nothrow @safe unittest // issue 15884
     assert(test(30738.22));
 
     assert(test(1 + double.epsilon));
-    assert(test(-double.max));
     assert(test(double.min_normal));
+    static if (realInDoublePrecision)
+        assert(test(-double.max / 2));
+    else
+        assert(test(-double.max));
```

Comment:
What is the result of this test when it fails? What consequences could the failure have for the API user?